agent/consul/leader_connect_ca.go

```diff
@@ -272,7 +272,7 @@ func newCARoot(pemValue, provider, clusterID string) (*structs.CARoot, error) {
 		ExternalTrustDomain: clusterID,
 		NotBefore:           primaryCert.NotBefore,
 		NotAfter:            primaryCert.NotAfter,
-		RootCert:            pemValue,
+		RootCert:            lib.EnsureTrailingNewline(pemValue),
```

Comment:
Older versions of Vault did not add a newline to the PEM file. `lib.EnsureTrailingNewline` is used in a few places in CA code to patch `\n` if missing.

My new test case revealed that the CARoot stored in CAManager did not have a trailing `\n`